systemscan/main.py

```diff
@@ -264,8 +264,16 @@ def legacy_inventory(inv):
     for iface in interfaces:
         iface_path = os.path.join('/sys/class/net', iface)
 
-        # don't count virtual devices (lo, bridge) or wifi devices
-        if ('virtual' in os.readlink(iface_path)) or os.path.exists(os.path.join(iface_path, 'phy80211')):
+        # skip if path is not a dir as later we relay on files inside it
+        if not iface.is_dir():
+            continue
+
+        # don't count virtual devices (lo, bridge)
+        if (iface.is_symlink() and 'virtual' in os.readlink(iface_path))
```

Comment:
Same as before, `iface` is a string and only the entry name, not the full path,
```suggestion
        if (os.path.islink(iface_path) and 'virtual' in os.readlink(iface_path))
```